liquibase-core/src/main/java/liquibase/LiquibaseRunner.java

```diff
@@ -0,0 +1,225 @@
+package liquibase;
+
+import liquibase.changelog.annotations.reader.ChangeEntry;
+import liquibase.changelog.annotations.reader.ChangeService;
+import liquibase.database.GenericDataStore;
+import liquibase.exception.LiquibaseChangeSetException;
+import liquibase.exception.LiquibaseConfigurationException;
+import liquibase.exception.LiquibaseException;
+import liquibase.logging.LogService;
+import liquibase.logging.Logger;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.env.Environment;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import static org.springframework.util.StringUtils.hasText;
+
+public class LiquibaseRunner implements InitializingBean {
+
+    private static final Logger logger = LogService.getLog(LiquibaseRunner.class);
+
+    private boolean enabled = true;
+    private String changeLogsScanPackage;
+    private ApplicationContext context;
+    private Environment springEnvironment;
+    private GenericDataStore genericDataStore;
+
+
+    public LiquibaseRunner(ApplicationContext context) throws ReflectiveOperationException {
+        this.context = context;
+        this.springEnvironment = context.getEnvironment();
+        String url = springEnvironment.getProperty("spring.liquibase.connectionURL");
+        String user = springEnvironment.getProperty("spring.liquibase.user");
+        String password = springEnvironment.getProperty("spring.liquibase.password");
+        String driver = springEnvironment.getProperty("spring.liquibase.driver");
+
+        //TODO: Validate url/user/password/driver
+        Class<?> c = Class.forName(driver);
+        Constructor<?> cons = c.getConstructor();
+        this.genericDataStore = (GenericDataStore) cons.newInstance();
+        this.genericDataStore.connect(url, user, password);
+    }
+
+
+    public LiquibaseRunner(String driver, String url, String user, String password) throws ReflectiveOperationException  {
+        Class<?> c = Class.forName(driver);
+        Constructor<?> cons = c.getConstructor();
+        this.genericDataStore = (GenericDataStore) cons.newInstance();
+        this.genericDataStore.connect(url, user, password);
+    }
+
+    /**
+     * For Spring users: executing liquicouch after bean is created in the Spring context
+     *
+     * @throws Exception exception
+     */
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        execute();
+    }
+
+    /**
+     * Executing migration
+     *
+     * @throws LiquibaseException exception
+     */
+    public void execute() throws LiquibaseException {
```

Comment:
The method that should be called for non-spring users during the application startup to start the migration. Are there any Liquibase class that could directly call it?

Reply:
The standard liquibase code has a separate class for a spring bean vs. re-using the Main class. We should be able to use that class as well